### PR TITLE
TextBox masks: get rid of legacy default strategy and refactor mask strategy

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

**Review by marker-dao — DISMISSED** [review]

File: js/ui/text_box/ui.text_editor.mask.strategy.js [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+
+const MASK_EVENT_NAMESPACE = 'dxMask';
+const BLUR_EVENT = 'blur beforedeactivate';
+const EMPTY_CHAR = ' ';
+const DELETE_INPUT_TYPES = ['deleteContentBackward', 'deleteSoftLineBackward', 'deleteContent', 'deleteHardLineBackward'];
+const HISTORY_INPUT_TYPES = ['historyUndo', 'historyRedo'];
+const EVENT_NAMES = ['focusIn', 'focusOut', 'input', 'paste', 'cut', 'drop', 'beforeInput'];
+
+function getEmptyString(length) {
+    return ' '.repeat(length);
```
Comment (alexanderPolosatov) [inline js/ui/text_box/ui.text_editor.mask.strategy.js]: let's replace ' ' to EMPTY_CHAR usage?

File: js/ui/text_box/ui.text_editor.mask.strategy.js [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+    _editorCaret(newCaret) {
+        if(!newCaret) {
+            return this.editor._caret();
+        }
+
+        this.editor._caret(newCaret);
+    }
+
+    _attachChangeEventHandler() {
+        if(!this._editorOption('valueChangeEvent').split(' ').includes('change')) {
```
Comment (alexanderPolosatov) [inline js/ui/text_box/ui.text_editor.mask.strategy.js:39]: let's replace ' ' to EMPTY_CHAR usage?

**Review by Zedwag — DISMISSED** [review]

File: js/ui/text_box/ui.text_editor.mask.strategy.js (hunk 2 above)
Comment (ksercs) [reply js/ui/text_box/ui.text_editor.mask.strategy.js:39]: space here means not an empty char, but a usual space)

File: js/ui/text_box/ui.text_editor.mask.strategy.js (hunk 1 above)
Comment (ksercs) [reply js/ui/text_box/ui.text_editor.mask.strategy.js]: done

**Review by alexanderPolosatov — APPROVED** [review]

**Pull request merged**